bigframes/core/__init__.py

```diff
@@ -49,470 +40,196 @@
 
 
 @dataclass(frozen=True)
-class WindowSpec:
-    """
-    Specifies a window over which aggregate and analytic function may be applied.
-    grouping_keys: set of column ids to group on
-    preceding: Number of preceding rows in the window
-    following: Number of preceding rows in the window
-    ordering: List of columns ids and ordering direction to override base ordering
-    """
-
-    grouping_keys: typing.Sequence[str] = tuple()
-    ordering: typing.Sequence[OrderingColumnReference] = tuple()
-    preceding: typing.Optional[int] = None
-    following: typing.Optional[int] = None
-    min_periods: int = 0
-
-
-# TODO(swast): We might want to move this to it's own sub-module.
 class ArrayValue:
```

Comment:
Maybe keep a docstring explaining this represents an ordered heterogenous data for now?

Reply:
added